src/marketplace/create_listings.ts

```diff
@@ -1,196 +1,277 @@
+      message: 'Enter the ID for the token to be listed',
+    });
+  }
 
-    if (options.projectAccessKey) {
-        nodeUrl += "/" + options.projectAccessKey
-    }
-    
-    if (options.verbose) {
-      console.log(`Using node URL: ${nodeUrl}`)
-    }
-  
-    const provider = new ethers.JsonRpcProvider(nodeUrl);
-    const wallet = new ethers.Wallet(privateKey, provider);
-  
-    console.log(`Using EOA Wallet: ${wallet.address}`);
-    console.log("Collection Address:", collectionAddress);
-    console.log("Currency token address:", currency);
-  
-    const collectionContract = new ethers.Contract(
-      collectionAddress,
-      ERC1155_ABI,
-      wallet
+  if (!network) {
+    console.log(
+      'Please provide the Network for your project as a Sequence chain handle'
+    );
+    console.log(
+      'Possible networks can be found at https://docs.sequence.xyz/solutions/technical-references/chain-support'
     );
 
-    const balance = await collectionContract.balanceOf(wallet.address, tokenId);
-  
-    let isApprovedForAll = false;
-  
+    network = await input({
+      message: 'Enter the network to be used (mainnet, polygon, etc.)',
+    });
+  }
+
+  if (!currency) {
+    currency = await input({
+      message:
+        'Enter either a well known currency code (USDC, MATIC, etc.) or a token address',
+    });
+  }
+
+  if (!price) {
+    price = await input({
+      message: 'Enter the price per token, in the currency specified',
+    });
+  }
+
+  const chainConfig = findSupportedNetwork(network);
+  if (chainConfig === undefined) {
+    program.error('Unsupported network, please select a valid network');
+    return;
+  }
+
+  let nodeUrl = chainConfig.rpcUrl;
+
+  if (options.projectAccessKey) {
+    nodeUrl += '/' + options.projectAccessKey;
+  }
+
+  if (options.verbose) {
+    console.log(`Using node URL: ${nodeUrl}`);
+  }
+
+  const provider = new ethers.JsonRpcProvider(nodeUrl);
+  const wallet = new ethers.Wallet(privateKey, provider);
+
+  // Set up the wallet client
+  const walletClient = createWalletClient({
+    account: privateKeyToAccount(privateKey),
+    chain: {
+      id: chainConfig.chainId,
+      name: chainConfig.name,
+      nativeCurrency: {
+        name: chainConfig.nativeToken.name,
+        decimals: chainConfig.nativeToken.decimals,
+        symbol: chainConfig.nativeToken.symbol,
+      },
+      rpcUrls: {
+        default: {
+          http: [nodeUrl],
+        },
+      },
+    },
+    transport: http(nodeUrl),
+  });
+
+  let walletAddress = wallet.address as Address;
+  if (!walletAddress.startsWith('0x')) {
+    walletAddress = `0x${walletAddress}`;
+  }
+
+  let marketplaceAddress = SEQUENCE_MARKETPLACE_V2_ADDRESS;
+  if (options.marketplaceVersion === 'v1') {
+    marketplaceAddress = SEQUENCE_MARKETPLACE_V1_ADDRESS;
+  }
+
+  let marketplaceABI = SequenceMarktetplace_V2_ABI;
+  if (options.marketplaceVersion == 'v1') {
+    let marketplaceABI = SequenceMarktetplace_V1_ABI;
```

Comment:
```suggestion
    marketplaceABI = SequenceMarktetplace_V1_ABI;
```

Reply:
ABI is a constant so I had to do it like this: 

```
  const marketplaceABI =
    options.marketplaceVersion == 'v2'
      ? SequenceMarketplace_V2_ABI
      : SequenceMarketplace_V1_ABI;
```